src/h5web/providers/hsds/utils.ts

```diff
@@ -26,3 +30,66 @@ export function assertHsdsDataset(
     throw new Error('Expected entity to be HSDS dataset');
   }
 }
+
+export function convertHsdsBaseType(hsdsBaseType: HsdsBaseType): HDF5BaseType {
+  if (hsdsBaseType.class === HDF5TypeClass.String) {
+    return hsdsBaseType;
+  }
+
+  const { class: hsdsClass, base } = hsdsBaseType;
+
+  const regex = /H5T_(?:STD|IEEE)_([A-Z])(\d+)(BE|LE)/u;
+  const matches = regex.exec(base);
+
+  if (!matches) {
+    throw new Error(`Unrecognized base ${base}`);
+  }
+
+  const [, sign, size, endianness] = matches;
+  assertEndianness(endianness);
```

Comment:
It's a case where it makes sense to do `endianness: endianness as HDF5Endianness`, because it cannot be anything else (unless there were to be a bug in the regex, in which case we'd fix it)